components/bootloaders/STM/stm32f103/include/HW.h

```diff
@@ -0,0 +1,265 @@
+    volatile uint32_t       SCR;
+    volatile uint32_t       CCR;
+    volatile uint32_t       SHPR[3];
+    volatile uint32_t       SHCSR;
+    volatile uint32_t       CFSR;
+    volatile uint32_t       HFSR;
+    volatile uint32_t       DFSR;
+    volatile uint32_t       MMFAR;
+    volatile uint32_t       BFAR;
+    volatile uint32_t       AFSR;
+} SCB_TypeDef;
+#define pSCB    ((struct SCB_TypeDef*)SCB_BASE)
+
+
+/* todo: there must be some major misunderstanding in how we access
+ * regs. The direct access approach (GET_REG) causes the usb init to
+ * fail upon trying to activate RCC_APB1 |= 0x00800000. However, using
+ * the struct approach from ST, it works fine...temporarily switching
+ * to that approach */
+typedef struct
+{
+    volatile uint32_t CR;
+    volatile uint32_t CFGR;
+    volatile uint32_t CIR;
+    volatile uint32_t APB2RSTR;
+    volatile uint32_t APB1RSTR;
+    volatile uint32_t AHBENR;
+    volatile uint32_t APB2ENR;
+    volatile uint32_t APB1ENR;
+    volatile uint32_t BDCR;
+    volatile uint32_t CSR;
+} RCC_RegStruct;
+#define pRCC                    ((RCC_RegStruct *)RCC_BASE)
+
+#define RCC_CR                  (RCC_BASE)
+#define RCC_CFGR                (RCC_BASE + 0x04)
+#define RCC_CIR                 (RCC_BASE + 0x08)
+#define RCC_AHBENR              (RCC_BASE + 0x14)
+#define RCC_APB2ENR             (RCC_BASE + 0x18)
+#define RCC_APB1ENR             (RCC_BASE + 0x1C)
+
+// Clocks for the backup domain registers
+#define RCC_APB1ENR_PWR_CLK     (0x10000000UL) // PWR clock enable
+#define RCC_APB1ENR_BKP_CLK     (0x08000000UL) // Backup periheral clock enable
+#define RCC_APB1ENR_CAN1_CLK    (0x02000000UL) // CAN peripheral clock enable
+#define RCC_APB2ENR_AFIO_CLK    (0x00000001UL) // Alternate Function I/O clock enable
+
+
+// Backup peripheral register map type.
+typedef struct bkp_reg_map
+{
+    const uint32_t    RESERVED1; // Reserved
+    volatile uint16_t DR1;       // Data register 1
+    const uint16_t    RESERVED2;
+    volatile uint16_t DR2;       // Data register 2
+    const uint16_t    RESERVED3;
+    volatile uint16_t DR3;       // Data register 3
+    const uint16_t    RESERVED4;
+    volatile uint16_t DR4;       // Data register 4
+    const uint16_t    RESERVED5;
+    volatile uint16_t DR5;       // Data register 5
+    const uint16_t    RESERVED6;
+    volatile uint16_t DR6;       // Data register 6
+    const uint16_t    RESERVED7;
+    volatile uint16_t DR7;       // Data register 7
+    const uint16_t    RESERVED8;
+    volatile uint16_t DR8;       // Data register 8
+    const uint16_t    RESERVED9;
+    volatile uint16_t DR9;       // Data register 9
+    const uint16_t    RESERVED10;
+    volatile uint16_t DR10;      // Data register 10
+    const uint16_t    RESERVED11;
+    volatile uint32_t RTCCR;     // RTC control register
+    volatile uint32_t CR;        // Control register
+    volatile uint32_t CSR;       // Control and status register
+} bkp_reg_map;
```

Comment:
Consistent naming

Reply:
yeah this stuff is all copy-pasted, I'll rename everything before it's ready to merge. More important to get it working right now though